javaslang/src/main/java/javaslang/collection/Tree.java

```diff
@@ -748,12 +771,12 @@ public boolean isEmpty() {
 
         @Override
         public boolean isLeaf() {
-            return children.isEmpty();
+            return size == 1;
         }
 
         @Override
         public int size() {
-            return 1 + children.foldLeft(0, (acc, child) -> acc + child.length());
+            return size;
```

Comment:
Because someone omitted this when writing those classes.

I think lazy init might be not the best case with recursive operations. The cost of calculating size when creating a single Node is nothing, but when you need to traverse whole tree and then compute size recursively for every Node it might be painful.

Anyway, the best argument I can come up with is:
Authors seem to have followed the same principle when creating other classes eg. `List`